src/polydb.jl

```diff
@@ -0,0 +1,376 @@
+# into ENV["POLYDB_TEST_URI"]
+# (used to connect to the github services container for testing)
+function get_db()
+   # we explicitly set the cacert file, otherwise we might get connection errors because the certificate cannot be validated
+   client = Mongoc.Client(get(ENV, "POLYDB_TEST_URI", "mongodb://polymake:database@db.polymake.org/?authSource=admin&ssl=true&sslCertificateAuthorityFile=$(cacert)"))
+   return Database(client["polydb"])
+end
+
+# returns a Polymake.Polydb.Collection instance with the given name
+# sections and collections in the name are connected with the '.' sign,
+# i.e. names = "Polytopes.Lattice.SmoothReflexive"
+Base.getindex(db::Database, name::AbstractString) = Collection{Polymake.BigObject}(db.mdb[name])
+
+# search a collection for documents matching the criteria given by d
+function Mongoc.find(c::Collection{T}, d::Dict=Dict(); opts::Union{Nothing, Dict}=nothing) where T
+   return Cursor{T}(Mongoc.find(c.mcol, Mongoc.BSON(d); options=(isnothing(opts) ? nothing : Mongoc.BSON(opts))))
+end
+
+function Mongoc.find(c::Collection{T}, d::Pair...) where T
+   return Cursor{T}(Mongoc.find(c.mcol, Mongoc.BSON(d...)))
+end
+
+# creating `BSON` iterators from the respective `Polymake.BigObject` iterator
+# and vice versa
+function Collection{T}(c::Collection) where T<:Union{Polymake.BigObject, Mongoc.BSON}
+   return Collection{T}(c.mcol)
+end
+
+function Cursor{T}(cursor::Cursor) where T<:Union{Polymake.BigObject, Mongoc.BSON}
+   return Cursor{T}(cursor.mcursor)
+end
+
+# returns a Polymake.BigObject from a Mongoc.BSON document
+function parse_document(bson::Mongoc.BSON)
+   str = Mongoc.as_json(bson)
+   return @pm common.deserialize_json_string(str)
+end
+
+# Iterator
+
+Base.IteratorSize(::Type{<:Cursor}) = Base.SizeUnknown()
+Base.eltype(::Cursor{T}) where T = T
+Base.IteratorSize(::Type{<:Collection}) = Base.SizeUnknown()
+Base.eltype(::Collection{T}) where T = T
+
+# default iteration functions returning `Polymake.BigObject`s
+function Base.iterate(cursor::Polymake.Polydb.Cursor{Polymake.BigObject}, state::Nothing=nothing)
+    next = iterate(cursor.mcursor, state)
+    isnothing(next) && return nothing
+    return Polymake.Polydb.parse_document(first(next)), nothing
+end
+
+Base.iterate(coll::Polymake.Polydb.Collection{Polymake.BigObject}) =
+    return iterate(coll, find(coll))
+
+function Base.iterate(coll::Polymake.Polydb.Collection{Polymake.BigObject}, state::Polymake.Polydb.Cursor)
+    next = iterate(state, nothing)
+    isnothing(next) && return nothing
+    doc, _ = next
+    return doc, state
+end
+
+# functions for `BSON` iteration
+Base.iterate(cursor::Cursor{Mongoc.BSON}, state::Nothing=nothing) =
+    iterate(cursor.mcursor, state)
+
+Base.iterate(coll::Collection{Mongoc.BSON}) =
+   iterate(coll.mcol)
+
+Base.iterate(coll::Collection{Mongoc.BSON}, state::Mongoc.Cursor) =
+   iterate(coll.mcol, state)
+
+#Info
+
+# returns an `Array{String, 1}` of the fields of a collection
```

Comment:
Since we have all of those comments (thanks!), maybe we should turn them into docstrings immediately?

Reply:
yes, i am planning a separate pull for the documentation, which includes docstrings, doctests and automation of generating the documentation within the CI. this will be right after wrapping `Map{String, String}` (which is already done, without the tests), but i think it would be much easier and cleaner if this pull request here was merged into master before.